### PR TITLE
send back delegate info as part of task response

### PR DESCRIPTION
**Pull request by vistaarjuneja**
# Commit Checklist

Thank you for creating a pull request! To help us review / merge this can you make sure that your PR adheres as much as possible to the following.

## The Basics

If you are adding new functionality, please provide evidence of the new functionality.

### CODE REVIEW

File: command/harness/dlite/cleanup.go [hunk 1]
```diff
@@ -45,5 +45,9 @@ func (t *VMCleanupTask) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		logr.WithError(err).Error("could not destroy VM")
 		httphelper.WriteJSON(w, failedResponse(err.Error()), httpFailed)
 	}
-	httphelper.WriteJSON(w, VMTaskExecutionResponse{CommandExecutionStatus: Success}, httpOK)
+	resp := VMTaskExecutionResponse{}
```
Comment (shubham149) [inline command/harness/dlite/cleanup.go:48]: resp := VMTaskExecutionResponse{
  CommandExecutionStatus: Success,
  DelegateMetaInfo: DelegateMetaInfo{} ...
}

File: command/harness/dlite/dlite.go [hunk 2]
```diff
@@ -50,7 +51,12 @@ func (c *dliteCommand) startPoller(ctx context.Context, tags []string) error {
 	// Client to interact with the harness server
 	client := delegate.New(c.env.Dlite.ManagerEndpoint, c.env.Dlite.AccountID, c.env.Dlite.AccountSecret, true)
 	p := poller.New(c.env.Dlite.AccountID, c.env.Dlite.AccountSecret, c.env.Dlite.Name, tags, client, r)
-	err := p.Poll(ctx, taskExecutors, taskInterval)
+	info, err := p.Register(ctx)
+	if err != nil {
+		return err
```
Comment (shubham149) [inline command/harness/dlite/dlite.go:56]: log the error

File: command/harness/dlite/init.go [hunk 3]
```diff
@@ -79,5 +79,7 @@ func (t *VMInitTask) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	resp.ServiceStatuses = serviceStatuses
 	resp.IPAddress = setupResp.IPAddress
 	resp.CommandExecutionStatus = Success
+	resp.DelegateMetaInfo.HostName = t.c.delegateInfo.Host
```
Comment (shubham149) [inline command/harness/dlite/init.go:82]: resp := VMTaskExecutionResponse{
  ServiceStatuses: serviceStatuses,
  ...
}

File: command/harness/dlite/cleanup.go (hunk 1 above)
Comment (vistaarjuneja) [reply command/harness/dlite/cleanup.go]: done

File: command/harness/dlite/dlite.go (hunk 2 above)
Comment (vistaarjuneja) [reply command/harness/dlite/dlite.go:56]: logged in response of start poller on L109

File: command/harness/dlite/init.go (hunk 3 above)
Comment (vistaarjuneja) [reply command/harness/dlite/init.go]: Done

**Review by shubham149 — APPROVED** [review]

**Pull request merged**